Yosemite/Yosemite/Stores/AnnouncementsStore.swift

```diff
@@ -88,34 +99,87 @@ private extension AnnouncementsStore {
     }
 
     /// Map `WordPressKit.Announcement` to `StorageAnnouncement` model
-    func mapAnnouncementToStorageModel(_ announcement: Announcement) -> StorageAnnouncement {
+    func mapAnnouncementToStorageModel(_ announcement: Announcement, displayed: Bool = false) -> StorageAnnouncement {
         let mappedFeatures = announcement.features.map {
             StorageFeature(title: $0.title,
                            subtitle: $0.subtitle,
                            iconUrl: $0.iconUrl,
                            iconBase64: $0.iconBase64)
         }
 
-        return StorageAnnouncement(appVersion: announcement.appVersionName,
-                                   features: mappedFeatures,
+        return StorageAnnouncement(appVersionName: announcement.appVersionName,
+                                   minimumAppVersion: announcement.minimumAppVersion,
+                                   maximumAppVersion: announcement.maximumAppVersion,
+                                   appVersionTargets: announcement.appVersionTargets,
+                                   detailsUrl: announcement.detailsUrl,
                                    announcementVersion: announcement.announcementVersion,
-                                   displayed: false)
+                                   isLocalized: announcement.isLocalized,
+                                   responseLocale: announcement.responseLocale,
+                                   features: mappedFeatures,
+                                   displayed: displayed)
     }
 
-    /// Save the `Announcement` to the appropriate file.
-    func saveAnnouncement(_ announcement: Announcement) throws {
+    /// Save the `Announcement` on disk if it's a new or a different announcement comparing to the existing one
+    func saveAnnouncementIfNeeded(_ announcement: Announcement) throws {
+        if let storageAnnouncement = storageAnnouncement {
+            guard !announcement.isSameVersion(storageAnnouncement) else {
+                throw AnnouncementsStorageError.announcementAlreadyExists
+            }
+        }
+
+        guard let fileURL = featureAnnouncementsFileURL else { throw AnnouncementsStorageError.unableToFindFileURL }
         let mappedAnnouncement = self.mapAnnouncementToStorageModel(announcement)
+        try fileStorage.write(mappedAnnouncement, to: fileURL)
+    }
+
+    /// Load the latest saved `Announcement`. Returns an Error if there is no saved announcement.
+    func loadSavedAnnouncement(onCompletion: (Result<(Announcement, IsDisplayed), Error>) -> Void) {
+        do {
+            onCompletion(.success((try savedAnnouncement(), storageAnnouncement?.displayed == true)))
```

Comment:
I do agree! but that's mainly because the WordPress model does't have such property and we don't want to leak the storageModel to the app. Therefore, grabbing this information from the saved model was the best plan I have so far!

Reply:
Yes but you can easily fix that by dissecting your `savedAnnouncement` method into two.

1. Have a function `loadStoredAnnouncement() -> StorageAnnouncement?` to load only the saved storage announcement
2. Have a function `mapStoredAnnouncement() -> Announcement` which I believe you already have
3. Modify this function to something like
```swift
func loadSavedAnnouncement(completion: Result<Announcement, Error> -> Void) {
	let storage = loadStoredAnnouncement()
	let readOnly = mapStoredAnnouncement(storage)
   onCompletion(readOnly, storage.displayed)
}
```

Does that makes sense?

Reply:
Looks interesting! I will manage to do that in a few 👍 
Thanks for the insight!

Reply:
Done! ✅